drivers/Aqara/aqara-lock/src/credential_utils.lua

```diff
@@ -0,0 +1,151 @@
+local capabilities = require "st.capabilities"
+local utils = require "st.utils"
+
+local remoteControlStatus = capabilities.remoteControlStatus
+local lockCredentialInfo = capabilities["stse.lockCredentialInfo"]
+
+local credential_utils = {}
+
+local HOST_COUNT = "__host_count"
+
+local function save_data(driver)
+  driver.datastore:save()
+  driver.datastore:commit()
+end
+
+local function update_remote_control_status(driver, device, added)
+  local host_cnt = device:get_field(HOST_COUNT) or 0
+  if added then
+    if host_cnt == 0 then
+      device:emit_event(remoteControlStatus.remoteControlEnabled('true', { visibility = { displayed = false } }))
+    end
+    host_cnt = host_cnt + 1
+  else
+    host_cnt = host_cnt - 1
```

Comment:
Is it possible for this `host_cnt` to go negative if the `HOST_COUNT` field is defaulted to 0?

Reply:
I added code to check host_cnt is 0.